kolena/_experimental/dataset/_dataset.py

```diff
@@ -83,10 +84,21 @@ def _infer_datatype_value(x: str) -> str:
     return DatapointType.TABULAR.value
 
 
-def _infer_datatype(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
-    if FIELD_LOCATOR in df.columns:
-        return df[FIELD_LOCATOR].apply(_infer_datatype_value)
-    elif FIELD_TEXT in df.columns:
+def add_datatype(df: pd.DataFrame) -> None:
```

Comment:
Marked public since this function is utilized by backend. This function opts to modify the passed DataFrame so the prefix parsing logic can be shared

Reply:
I kinda disagree with this. This additional info is for internal use only, we don't expose this to users or want them to think about it. When users get retrieve the data back, they also don't see this temporary internal column.